lib/Service/AssistantService.php

```diff
@@ -66,50 +104,308 @@ public function sendNotification(TextProcessingTask|TextToImageTask $task, ?stri
 		$manager->notify($notification);
 	}
 
+	/**
+	 * @param string $writingStyle
+	 * @param string $sourceMaterial
+	 * @return string
+	 */
+	private function formattedCopywriterPrompt(string $writingStyle, string $sourceMaterial): string {
+		return "You're a professional copywriter tasked with copying an instructed or demonstrated *WRITING STYLE* and writing a text on the provided *SOURCE MATERIAL*. \n*WRITING STYLE*:\n$writingStyle\n\n*SOURCE MATERIAL*:\n\n$sourceMaterial\n\nNow write a text in the same style detailed or demonstrated under *WRITING STYLE* using the *SOURCE MATERIAL* as source of facts and instruction on what to write about. Do not invent any facts or events yourself. Also, use the *WRITING STYLE* as a guide for how to write the text ONLY and not as a source of facts or events.";
+	}
+
+	/**
+	 * Sanitize inputs for storage based on the input type
+	 * @param string $type
+	 * @param array $inputs
+	 * @return array
+	 * @throws \Exception
+	 */
+	private function sanitizeInputs(string $type, array $inputs): array {
+		switch ($type) {
+			case 'copywriter':
+				{
+					// Sanitize the input array based on the allowed keys and making sure all inputs are strings:
+					$inputs = array_filter($inputs, function ($value, $key) {
+						return in_array($key, ['writingStyle', 'sourceMaterial']) && is_string($value);
+					}, ARRAY_FILTER_USE_BOTH);
+
+					if (count($inputs) !== 2) {
+						throw new \Exception('Invalid input(s)');
+					}
+					break;
+				}
+			default:
+				{
+					if (!is_string($inputs['prompt']) || count($inputs) !== 1) {
+						throw new \Exception('Invalid input(s)');
+					}
+					break;
+				}
+		}
+		return $inputs;
+	}
+
 	/**
 	 * @param string|null $userId
 	 * @param int $taskId
-	 * @return TextProcessingTask
+	 * @return Task
 	 */
-	public function getTextProcessingTask(?string $userId, int $taskId): ?TextProcessingTask {
+	public function getTextProcessingTask(?string $userId, int $taskId): ?Task {
 		try {
-			$task = $this->textProcessingManager->getTask($taskId);
-		} catch (NotFoundException | \RuntimeException $e) {
+			$task = $this->taskMapper->getTask($taskId);
+		} catch (DoesNotExistException | MultipleObjectsReturnedException | \OCP\Db\Exception $e) {
 			return null;
 		}
 		if ($task->getUserId() !== $userId) {
 			return null;
 		}
+		// Check if the task status is up-to-date (if not, update status and output)
+		try {
+			$ocpTask = $this->textProcessingManager->getTask($task->getOcpTaskId());
+
+			if($ocpTask->getStatus() !== $task->getStatus()) {
+				$task->setStatus($ocpTask->getStatus());
+				$task->setOutput($ocpTask->getOutput());
+				$task = $this->taskMapper->update($task);
+			}
+		} catch (NotFoundException $e) {
+			// Ocp task not found, so we can't update the status
```

Comment:
It's conceivable that an assistant meta task is stored for longer than the corresponding ocp task. This would be the case, for example, with generated images, if we unify the routes and front-end for all task categories at some point in the future. That's why I initially considered the generic catching of all exceptions acceptable here, since the primary task of the method has already been completed (although it definitely makes sense to do some debug logging).